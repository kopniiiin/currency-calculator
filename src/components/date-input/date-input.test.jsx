import React from "react";
import {shallow} from "enzyme";

import DateInput from "./date-input.jsx";

describe(`snapshot test: DateInput`, () => {
  it(`should render correctly`, () => expect(
    shallow(
      <DateInput
        name={`date`}
        value={`2020-04-04`}
        label={`date`}
        onChange={() => {}}/>
    )
  ).toMatchSnapshot());
});

describe(`e2e test: DateInput`, () => {
  it(`should call onChange with value`, () => {
    const value = `2020-06-06`;
    const onChange = jest.fn();

    shallow(
      <DateInput
        name={`date`}
        value={`2020-04-04`}
        label={`date`}
        onChange={onChange}/>
    ).find(`.date-input`).simulate(`change`, {target: {value}});

    expect(onChange).toHaveBeenCalledTimes(1);
    expect(onChange.mock.calls[0][0]).toBe(value);
  });
});
