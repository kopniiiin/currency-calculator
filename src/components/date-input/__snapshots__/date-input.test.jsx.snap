// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot test: DateInput should render correctly 1`] = `
<input
  aria-label="date"
  className=" date-input"
  max=""
  name="date"
  onChange={[Function]}
  type="date"
  value="2020-04-04"
/>
`;
